scripts/generate-new-orb-id/gen-orb-id.sh

```diff
@@ -0,0 +1,281 @@
+    install -o 1000 -g 1000 -m 664 "${BUILD_DIR}/calibration.json" "${mount_target}/calibration.json"
+    install -o 1000 -g 1000 -m 664 "${BUILD_DIR}/versions.json" "${mount_target}/versions.json"
+    install -o 1000 -g 1000 -m 664 /dev/null "${mount_target}/hardware_version"
+    echo "${hardware_version}" > "${mount_target}/hardware_version"
+    setfacl -d -m u::rwx,g::rwx,o::rx "${mount_target}"
+    sync
+    umount "${mount_target}"
+}
+
+register_orb() {
+    local orb_id="$1"
+    local bearer="$2"
+    local release_type="$3"
+    local hardware_version="$4"
+
+    local orb_name
+    orb_name="$(cat "${ARTIFACTS_DIR}/${orb_id}/orb-name")"
+
+    local is_dev
+    case "${release_type}" in
+        "dev")
+            is_dev="true" ;;
+        "prod")
+            is_dev="false" ;;
+        *)
+            echo "Error: Invalid release type specified."
+            usage; exit 1 ;;
+    esac
+
+    curl --fail --location --request POST "${CORE_APP_REGISTRATION_URL}" \
+        --header "Authorization: Bearer ${bearer}" \
+        --header 'Content-Type: application/json' \
+        --data-raw '{
+            "query":"mutation InsertOrb($deviceId: String, $name: String!) { insert_orb(objects: [{name: $name, deviceId: $deviceId, status: FLASHED, deviceType: '"${hardware_version}"', isDevelopment: '"${is_dev}"'}], on_conflict: {constraint: orb_pkey}) {affected_rows}}",
+            "variables": {"deviceId": "'"${orb_id}"'", "name": "'"${orb_name}"'"}
+        }' | jq -re 'if .data.insert_orb.affected_rows == 1 then true else error("Failed to register Orb") end'
+}
+
+main() {
+    local bearer="${FM_CLI_ORB_MANAGER_INTERNAL_TOKEN:-}"
+    local hardware_token="${HARDWARE_TOKEN_PRODUCTION:-}"
+    local hardware_version="${HARDWARE_VERSION:-}"
+    local backend="${FM_CLI_ENV:-}"
+    local release_type
+
+    local arg
+    local num_ids
+    local positional_args=()
+    while [[ $# -gt 0 ]]; do
+        arg="$1" ; shift
+        case "${arg}" in
+            -h|--help)
+                usage; exit 0 ;;
+            -t|--token)
+                bearer="${1}"; shift ;;
+            -b|--backend)
+                backend="${1}"; shift ;;
+            -r|--release)
+                release_type="${1}"; shift ;;
+            -v|--hardware-version)
+                hardware_version="${1}"; shift ;;
+            -c|--registration-token)
+                hardware_token="${1}"; shift ;;
+            -*)
+                echo "Unknown option: ${1}"
+                usage; exit 1 ;;
+            *)
+                positional_args+=("${arg}") ;;
+        esac
+    done
+
+    if [[ ${#positional_args[@]} -ne 1 ]]; then
+        echo "Error: <id_num> is required."
+        usage
+        exit 1
+    fi
+    num_ids="${positional_args[0]}"
+
+    if [[ -z "${bearer}" || -z "${hardware_token}" || -z "${hardware_version}" || -z "${backend}" || -z "${release_type}" ]]; then
```

Comment:
I'd remove `release_type` from here and use the explicit separate check above. Right now, this will fail since it's null / unset. Better to group by how they're being treated.